proposals/005_New_Security_Definitions.md

```diff
@@ -0,0 +1,531 @@
+certain circumstances:
+
+* `type: apiKey`: This will allow you to provide your JWT anywhere you would like, but it could be confusing to overload
+  `type: apiKey` since you are describing a Security Scheme using a JWT and not an API Key
+* `type: http`: This will only work if your JWT is in a header and in a supported
+  [HTTP Auth](https://datatracker.ietf.org/doc/html/rfc7235#section-5.1) format
+
+OpenAPI being so opinionated about these kinds things does not serve any real benefit to anyone involved, and this
+proposal would make things much more flexible.
+
+## Proposed solution
+
+This proposal suggests the following changes to the Security Scheme object:
+
+* Add a new `config` property that is type specific to contain the scheme-specific configuration not realted to
+  credential configuration
+* Add a new `credentials` property that is an array of [Credential Objects](#credential-object) that describe a
+  credential _(where it is located, what its format is, etc.)_
+* Remove the credential-specific properties _(`bearerFormat`, `flows`, `in`, `openIdConnectUrl` and `scheme`)_ and move
+  them into a type-specific `config` property
+* Update `type` to allow for any value instead of the hard-coded list of supported types
+
+## Detailed design
+
+The bulk of this proposal is defining a way to separate the properties used to describe the credential(s) for a Security
+Scheme from the root properties of the Security Scheme.  This proposal will also allow you to define multiple
+credentials _(multiple locations, formats, ...)_ for the same Security Scheme for cases where you might support
+multiple formats/locations for effectively the same credential.  The new objects being suggested as part of this
+proposal are defined below.
+
+**Note:** While not specifically mentioned below for brevity, JSON References and Extensions should be allowed within
+every complex type documented below.
+
+### Value Getter Object
+
+The _Value Getter Object_ describes a place in the request where something exists that you want to retrieve.  The name
+and purpose of this object is generic, as in I could see other places in OpenAPI where you could point to a value in the
+request.  The _Value Getter Object_ is the base object to be _extended_ by **all**
+[Credential Object](#credential-object)s.  Below is its schema:
+
+| Field Name | Type | Applies To | Description |
+| ---------- | ---- | ---------- | ----------- |
+| in | `string` | Any | **REQUIRED.** Specifies where in the request the credential is provided.  Valid values are: `body`, `cookie`, `header`, `path` and `query` |
+| format | `regex` | Any | Specifies the format of the credential.  Since in many cases, the credential is a subset of the raw value, `format` must be a regular expression that matches the complete value, and contains a single capturing group to match the actual value within the raw value. |
```

Comment:
Rather than express how to **retrieve** the credential using a regex, it may be more OAS-like to provide a *template* for how the credential should be constructed. Thus 

> `format: ^[B|b][E|e][A|a][R|r][E|e][R|r] (.*)$`

would become something like:

> `format: Bearer {credential}`

PS: Header names are case-insensitive, but the "Bearer" string is not, so this regex can be simplified.

Reply:
I don't disagree.  The main reason for using a regex example was to show how one might pick out a portion of a larger HTTP value, I'm not married to any specific implementation.